src/Meziantou.Analyzer/Rules/AvoidComparisonWithBoolLiteralAnalyzer.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Meziantou.Analyzer.Rules
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class AvoidComparisonWithBoolLiteralAnalyzer : DiagnosticAnalyzer
+    {
+        private static readonly DiagnosticDescriptor s_rule = new DiagnosticDescriptor(
+            RuleIdentifiers.AvoidComparisonWithBoolLiteral,
+            title: "Avoid comparison with bool literal",
+            messageFormat: "Avoid comparison with bool literal",
+            RuleCategories.Style,
+            DiagnosticSeverity.Info,
+            isEnabledByDefault: true,
+            description: "",
+            helpLinkUri: RuleIdentifiers.GetHelpUri(RuleIdentifiers.AvoidComparisonWithBoolLiteral));
+
+        private static readonly ImmutableArray<SyntaxKind> s_comparisonExpressionKinds =
+            ImmutableArray.Create(SyntaxKind.EqualsExpression, SyntaxKind.NotEqualsExpression);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(s_rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSyntaxNodeAction(AnalyzeComparisonWithBool, s_comparisonExpressionKinds);
```

Comment:
For this analyzer, I would go with `IBinaryOperation`. This way you can get the type of the operands and be sure one operand if of type `bool` and the other one is a constant (true or false).

If you are not aware of IOperation, you can check my blog post: <https://www.meziantou.net/writing-a-language-agnostic-roslyn-analyzer-using-ioperation.htm>. You can also check other analyzers that use operations.

Reply:
Made changes to the Analyzer (but not the Fixer). Let me know what you think.

Reply:
In the end, I had to also modify the fixer.

Reply:
That's good :)